kale/codegen/generate_code.py

```diff
@@ -160,12 +164,18 @@ def generate_pipeline(template, nb_graph, step_names, lightweight_components,
     leaf_steps = [x for x in nb_graph.nodes()
                   if nb_graph.out_degree(x) == 0]
 
+    # create a dict with step names and their parameters
+    steps_parameters = {step: sorted(nb_graph.nodes(data=True)[step]
```

Comment:
Rename suggestion: `all_step_parameters`, or something else if you have a preference.

I'm not fond of `step_parameters` and `steps_parameters`, because the difference between the names is pretty small and it's not clear what the difference is. Thus, it's a bug trap :sweat_smile: